### PR TITLE
Script case validation

### PR DESCRIPTION
**Pull request by davidsbailey**
Levelbuilders on /s/K5-OnlinePD were getting errors saving changes to scripts with upper case names: https://app.honeybadger.io/projects/3240/faults/45403228

The root cause was a name validation failure here: https://github.com/code-dot-org/code-dot-org/blob/cb2fd67b5812161aa7e809f33fd3a3555c4ff747/dashboard/app/models/script.rb#L1089-L1093

Name validation didn't make sense here, since we aren't modifying the name. So I disabled name validation at this location.

This fixes saving /s/K5-OnlinePD for me locally.

### CODE REVIEW

File: apps/src/lib/script-editor/ScriptEditor.jsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export default class ScriptEditor extends React.Component {
     loginRequired: PropTypes.bool,
     hideableStages: PropTypes.bool,
     studentDetailProgressView: PropTypes.bool,
-    professionalLearningCourse: PropTypes.bool,
+    professionalLearningCourse: PropTypes.string,
```
Comment (davidsbailey) [inline apps/src/lib/script-editor/ScriptEditor.jsx:40]: this is just a bonus prop type fix in response to a console warning I saw when editing the script on levelbuilder.

**Review by Hamms — APPROVED** [review]
Is the problem that these scripts have invalid names (but we want to keep them around anyway), or that our name validation rules are overly strict?

This seems like a good change to unblock levelbuilders, but we should probably also address this in a more systemic way

**Comment by davidsbailey** [comment]
The problem is that some of these scripts have invalid names. I'm not sure we'll ever want to rename these scripts with invalid names, so grandfathering these in while disallowing creation of new scripts with invalid names seems like the best option I can think of. I'm open to a more systemic approach as a follow-up, though. Any thoughts on what that would look like?

**Comment by Hamms** [comment]
Do we have a specific reason for enforcing the naming scheme that we do?

**Comment by davidsbailey** [comment]
Good question! Some context here when it was added: https://github.com/code-dot-org/code-dot-org/pull/9625 My gut feel is that it's not worth changing this without a compelling reason, though.

**Comment by davidsbailey** [comment]
Merging this now to unblock the levelbuilders, but I'm open to more follow up as needed.

**Pull request merged**